speaker/layer2_controller.go

```diff
@@ -66,10 +66,11 @@ func usableNodes(eps k8s.EpsOrSlices, speakers map[string]bool) []string {
 				if !k8s.IsConditionReady(ep.Conditions) {
 					continue
 				}
-				nodeName := ep.Topology["kubernetes.io/hostname"]
-				if nodeName == "" {
+				nodeNamePtr := ep.NodeName
```

Comment:
This can be reduced to:
```
if ep.NodeName == nil {
   continue
}
```

and using `*ep.NodeName` below in place of nodeName

Reply:
Right, done :+1: I was following old logic.